src/Cache/CacheAdapter.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Skautis\Nette\Cache;
+
+use DateInterval;
+use Exception;
+use Nette\Caching\Cache;
+use Psr\SimpleCache\CacheInterface;
+use Tracy\Debugger;
+use Traversable;
+
+
+/**
+ * Nette cache adapter for Skautis library
+ */
+class CacheAdapter
+  implements
+  CacheInterface
+{
+
+  /**
+   * @var Cache
+   */
+  private $cache;
+
+  /**
+   * @var int|null
+   */
+  private $defaultTTL;
+
+  public function __construct(
+    Cache $cache,
+    $defaultTTLSeconds = null
+  ) {
+    $this->cache = $cache;
+    $this->defaultTTL = $defaultTTLSeconds;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public function get(
+    $key,
+    $default = null
+  ) {
+    $this->assertValidKey($key);
+
+    try {
+      $value = $this->cache->load(
+        $key,
+        static function () use
+        (
+          $default
+        ) {
+          return $default;
+        }
+      );
+    } catch (Exception $exception) {
+      throw new CacheException("Failed to load key '$key''", $exception);
+    }
+
+    return $value;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public function set(
+    $key,
+    $value,
+    $ttl = null
+  ): bool {
+    $this->assertValidKey($key);
+
+    try {
+      $this->cache->save(
+        $key,
+        $value,
+        [
+          Cache::EXPIRE => $this->convertTTLToExpire($ttl),
+          Cache::SLIDING => false,
+        ]
+      );
+    } catch (Exception $exception) {
+      Debugger::log($exception);
```

Comment:
Tohle bude vyžadovat přidání Tracy jako přímé závislosti (místo suggest, jako je to teď).